functorch/_src/pytree_hacks.py

```diff
@@ -4,38 +4,7 @@
 # This source code is licensed under the BSD-style license found in the
 # LICENSE file in the root directory of this source tree.
 
-import torch.utils._pytree as _pytree
-from torch.utils._pytree import tree_flatten, tree_unflatten, TreeSpec
-from typing import List, Any
-
-# TODO: The following function should only be used with vmap.
-# torch.return_types should be registered as PyTree nodes.
-# I can't figure out how to do that, so we are turning all of them
-# into normal Tuples for now (this is what vmap used to do anyways).
-# We probably want some special behavior for named tuples?
-
-
-def tree_flatten_hack(pytree):
-    if _pytree._is_leaf(pytree) and not isinstance(pytree, tuple):
-        return [pytree], _pytree.LeafSpec()
```

Comment:
I don't remember if this hack was for namedtuples or torch.return_types or both. Could we add a test for functions that accept and return namedtuples?

Reply:
I don't think there are operators which return `namedtuples`. I mostly refer to `torch.return_types` as `namedtuples` probably because the [PR](https://github.com/pytorch/pytorch/pull/15429) that added them referred them as such.

Do let me know if you know any such operator.
Thanks! :)

Reply:
Hmm my suggestion was to test vmap over a function that returns a namedtuple. The function doesn't have to be an operator:
```
Point = namedtuple('Point', ['x', 'y'])

def f(x, y):
  return Point(x=x, y=y)

x = torch.randn(2, 5)
y = torch.randn(2, 3)
result = vmap(f)(x, y)
# check that result is a Point
```

Reply:
Updated.  Thanks :)!